Feature/milestonev2

aracelicaes opened the pull request:
TSE, please ensure you have Ruby Version 2.7.1 Installed as well as Rails Version 6.0.0
Run bundle install since we updated project gem versions.
In this Branch, we added the Friendship Model with the 2 foreign keys user_id and friend_is (that makes reference to the User id from the User Class).
We then added the associations of the Friendship Model inside the User Model, along with the methods that validate friendship request, pending request, confirmed request, and the one that evaluates if that friendship exists (#friends?(user)).
We followed this tutorial:
https://explainextended.com/2009/03/07/selecting-friends/

Please let us know if you have any questions.
Jess and Ara.

aracelicaes committed:
Milestone 1
Created ERD for Social Media Clone
Uploading jpeg

aracelicaes committed:
Edit Readme w/model img
Changed file name for model jpeg
Add img on Readme

jessicafarias committed:
Merge pull request #1 from jessicafarias/feature/milestone1

Feature/milestone1

jessicafarias committed:
Create friendship Model and db:migrate

aracelicaes committed:
Add User Associations and Methods
Add friendships association in User
Add inverse_friendships association in User
Add friends method in User
Add #pending_friends in User
Add #friend_requests in User
Add #confirm_friend in User
Add #friend?(*) in User

aracelicaes committed:
Updated ReadMe with Ruby and Rails new versions

jessicafarias committed:
Fixing rubocop errors

jessicafarias committed:
Adding friendships controller

Adding create verb for friendships
Adding friendships routes
adding requested_friend def for user models
Adding fields to view friendship status

aracelicaes committed:
Add #invitee to User Ctrlr
Add #invitee?(user) to User Ctlr
Updated Links in User #index views
Updated friendship ctrlr

jessicafarias committed:
Adding accept link

Edit update in friendship controller
Edit user model
Adding Friend confirm label

aracelicaes committed:
Adding #destroy in User ctrl
Adding #destroy method in User Ctrlr
Adding #destroy action in Friendship ctlrlr
Adding Delete button to user views
Adding decline button to user views
Adding conditional to user ctlr to only view other users not the current user.

aracelicaes committed:
Created Helpers
Created #friendship_buttons helper
Created #display_users helper
Added helpers to user/show and user/index
Corrected rubocop
Added bootstrap in gemfile
imported bootstrap in custom.scss

jessicafarias committed:
Adding rspec files

Adding testing for friendship model
Adding test for post model
Adding test for user model
Adding rspec in Gemfile

jessicafarias committed:
Adding rspec and styles

Merge branch 'feature/milestonev1' of github.com:jessicafarias/ror-social-scaffold into feature/milestonev1

aracelicaes committed:
Added a couple more tests to Post
Added Validations tests to post

jessicafarias committed:
Change the colors of the buttons

jessicafarias committed:
Adding styles and rspect test

Merge branch 'feature/milestonev1' of github.com:jessicafarias/ror-social-scaffold into feature/milestonev1

aracelicaes committed:
Delete bckcolor Add mr-2 to Accept button

aracelicaes committed:
added line to readme

BrittanyBlake reviewed (changes requested):
### Status: Changes Required ♻️ 
****
### Hi Team! 👋 
 You have done a wonderful job with your project so far!👏 👏 👏  There are just a few issues that need to be resolved before you can move on: ⬇️ 
****

- [x] The user should only see their posts and posts from their friends on their timeline.
Screenshot of requirement 👇 
![Screen Shot 2020-09-28 at 9 05 02 PM](https://user-images.githubusercontent.com/48804258/94474696-49530380-01ce-11eb-837b-ca3935b710b3.png)
Screenshot from your page 👇 
![Screen Shot 2020-09-28 at 9 05 30 PM](https://user-images.githubusercontent.com/48804258/94474737-58d24c80-01ce-11eb-8d22-7cbb4ef4c1f6.png)
![Screen Shot 2020-09-28 at 9 05 52 PM](https://user-images.githubusercontent.com/48804258/94474772-65ef3b80-01ce-11eb-85cb-977998485b04.png)
- As you can see, the user is not friends with TSE, yet the user can see the post. 

- [x] Your add friends button is messed up

![Screen Shot 2020-09-28 at 9 07 23 PM](https://user-images.githubusercontent.com/48804258/94474892-9cc55180-01ce-11eb-901c-bbe8b0d50a9e.png)

- [x] When the friendship is confirmed,  an additional row with reversed user-friend pair should be created
Screenshot from the console 👇 
![Screen Shot 2020-09-28 at 9 10 34 PM](https://user-images.githubusercontent.com/48804258/94475221-0d6c6e00-01cf-11eb-9501-04bb6f25f6aa.png)
- There is only 1 row instead of 2 as indicated in the requirements. This is so we can find user friends only by user_id in friendships table.
Screenshot from the requirements 👇 
![Screen Shot 2020-09-28 at 9 14 36 PM](https://user-images.githubusercontent.com/48804258/94475614-a00d0d00-01cf-11eb-9b48-8f4a44eb47d7.png)


****
### Please make the required changes and resubmit for a new code review once you are ready 🙏

Good Luck and Happy Coding 🍀 👩🏾‍💻

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Brittany Blake)._

aracelicaes committed:
Editing Model

jessicafarias committed:
Adding a new row of friendships when confirm friendship

Adding new row in Controller

aracelicaes committed:
Modified User Model methods

jessicafarias committed:
Adding delete/create rows

Create confirm_friend in user model
Create delete_friend in user model
Deleted comments

aracelicaes committed:
Refactored methods & fixed button
Refactored methods in User model
Fixed Add Friend Button in user/show.html
Deleted comments only left the essential

jessicafarias committed:
Restricted posts of current_user

MarioDena reviewed (approved):
Hello Team great work implementing the requested changes. 👍 

# Project approved! 🟢 

Cheers! 🥇
